Client/mods/deathmatch/logic/CStaticFunctionDefinitions.cpp

```diff
@@ -3603,6 +3603,77 @@ bool CStaticFunctionDefinitions::SetElementCollidableWith ( CClientEntity & Enti
 }
 
 
+bool CStaticFunctionDefinitions::SetElementCollidableWithType ( CClientEntity & Entity, const char* szTypeName, bool bCanCollide, bool bOnlyWithCreated )
+{
+    switch ( Entity.GetType () )
+    {
+        case CCLIENTPLAYER:
+        case CCLIENTPED:
+        case CCLIENTOBJECT:
+        case CCLIENTVEHICLE:
+        {
+            unsigned int uiType = CClientEntity::GetTypeID ( szTypeName );
+
+            switch ( uiType )
+            {
+                case CCLIENTPLAYER:
+                case CCLIENTPED:
+                case CCLIENTOBJECT:
+                case CCLIENTVEHICLE:
+                {
+                    // Change CollidableWith for all elements of this type on the server
+                    CFastList < CClientEntity* > entities = CClientEntity::GetEntitiesFromRoot ( HashString ( szTypeName ), false );
+
+                    CChildListType::const_iterator iter = entities.begin ();
+                    for ( ; iter != entities.end (); iter++ )

```

Comment:
I am not certain if you can use a for-range loop here, but **.begin()** and **.end()** seem promising

Reply:
Should I change GetEntitiesFromRoot return to list instead of CFastList?

Reply:
You should rather wait for somebody else to comment or benchmark both variants and keep the better one, because I can't answer this question.